### ISSUE TITLE
Need to implement request limits for http2

### ISSUE THREAD

**Issue by Bhashinee** [ballerina-platform/ballerina-standard-library#489]
**Description:**
Request limits set to maxUriLength, maxHeaderSize and maxEntityBodySize do not work with HTTP2.

**Comment by dilanSachi** [comment]
At the moment, this is not working with nghttp client and ballerina client with `http2PriorKnowledge`.
Even if we set the request limits, the default values by netty are considered as the limits.
Following are the actual limits which the server respond with error.

```
maxUriLength - 4K
maxHeaderSize - 8K
maxEntityBodySize - -1
```

However, once we exceed the maxUriLength, the server responds with `431 - Request Header Fields Too Large` rather than `414 - URI Too Long`
